s3fs/core.py

```diff
@@ -360,13 +373,35 @@ async def set_session(self, refresh=False, kwargs={}):
                 if key not in drop_keys
             }
             config_kwargs["signature_version"] = UNSIGNED
+
         conf = AioConfig(**config_kwargs)
         self.session = aiobotocore.AioSession(**self.kwargs)
-        s3creator = self.session.create_client(
-            "s3", config=conf, **init_kwargs, **client_kwargs
+
+        for parameters in (config_kwargs, self.kwargs, init_kwargs, client_kwargs):
```

Comment:
Can the region/endpoint really be in any one of these kwarg sets?

Reply:
from what I can see, all of them can contain `region_name`. And `self.kwargs` can contain `endpoint_url` so I just included it in this logic (since if I flatten all of these `not (config_kwargs.get("xxx") or config_kwargs.get("yyy") or self.kwargs.get("xxx") or ...)` they become really complicated).

Reply:
OK, I suppose these are all typically small dicts, so this won't take long.

Reply:
Also note that we are iterating on container objects themselves, not their contents. So at most we iterate over 4 objects and perform 2 lookups which is the same with 8X dict.get() calls, which is extremely cheap